clients/venice-push-job/src/main/java/com/linkedin/venice/hadoop/VenicePushJob.java

```diff
@@ -2271,6 +2303,19 @@ private StoreSetting getSettingsFromController(ControllerClient controllerClient
     storeSetting.isWriteComputeEnabled = storeResponse.getStore().isWriteComputationEnabled();
     storeSetting.isIncrementalPushEnabled = storeResponse.getStore().isIncrementalPushEnabled();
     storeSetting.storeRewindTimeInSeconds = DEFAULT_RE_PUSH_REWIND_IN_SECONDS_OVERRIDE;
+    if (setting.isTargetedRegionPushEnabled && setting.targetedRegions == null) {
+      // only override the targeted regions if it is not set and it is a single region push
+      setting.targetedRegions = storeResponse.getStore().getNativeReplicationSourceFabric();
+      if (StringUtils.isEmpty(setting.targetedRegions)) {
+        throw new VeniceException(
+            "The store either does not have native replication mode enabled or set up default source fabric.");
+      }
+      if (StringUtils.isNotEmpty(setting.targetedRegions)) {
```

Comment:
most of the time this will trigger as all stores are on NR today. Can we remove the inc_push + target region combo validation earlier at line ~691.

Reply:
Line 691 is still in the constructor where we haven't made the call to controller yet. That's why I moved here.

Reply:
I have no idea why I didn't take this suggestion but somehow revisit it again today, I think it makes sense to move up to 691 since `TARGETED_REGION_PUSH_ENABLED` should be set to true regardless the value of `TARGETED_REGION_PUSH_LIST`. I will add this to my next commit!